src/AstRelation.h

```diff
@@ -123,9 +123,12 @@ class AstRelation : public AstNode {
     /** Print string representation of the relation to a given output stream */
     virtual void print(std::ostream &os) const {
         os << ".decl " << this->getName() << "(";
-        os << attributes[0]->getAttributeName() << ":" << attributes[0]->getTypeName();
-        for (size_t i=1; i<attributes.size(); ++i) {
-            os << "," << attributes[i]->getAttributeName() << ":" << attributes[i]->getTypeName();
+        if (attributes.size() > 0) {
```

Comment:
this part could be reduced to a single line by using the `join` utility, which lists elements with a seperator